src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -1108,13 +1108,12 @@ export function calculateLayout(
                 throw (new Error("cannot find matching unique key in the grid layout"));
             }
         } else {
-            const visibleAttributeMap = _.map(visibleAttributes, attribute => attribute.uniqueKey);
-            currentGridLayout.forEach(chartLayout => {
-                //add only visible charts
-                if (chartLayout.i && visibleAttributeMap.includes(chartLayout.i)) {
-                    layout.push(chartLayout);
-                    availableChartLayoutsMap[chartLayout.i] = true;
-                }
+            // order charts based on x and y
+            const chartOrderMap = _.keyBy(currentGridLayout, chartLayout => chartLayout.i);
+            visibleAttributes.sort((a, b) => {
+                const chart1 = chartOrderMap[a.uniqueKey];
+                const chart2 = chartOrderMap[b.uniqueKey];
+                return (chart1 || chart2) ? (!chart2 ? -1 : !chart1 ? 1 : (chart1.y == chart2.y ? chart1.x - chart2.x : chart1.y - chart2.y)) : 0;
```

Comment:
i find multiple ternary too hard to read.    also, lets put a comment in explaining logic.